ecs-init/volumes/ecs_volume_driver.go

```diff
@@ -0,0 +1,238 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package volumes
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/docker/go-plugins-helpers/volume"
+)
+
+const (
+	// VolumeMountPathPrefix is the host path where amazon ECS plugin's volumes are mounted
+	VolumeMountPathPrefix = "/var/lib/ecs/volumes/"
+	// FilePerm is the file permissions for the host volume mount directory
+	FilePerm = 0700
+)
+
+// AmazonECSVolumeDriver holds list of volume drivers and volumes information
+type AmazonECSVolumeDriver struct {
+	volumeDrivers map[string]ECSVolumeDriver
+	volumes       map[string]*Volume
+	lock          sync.RWMutex
+}
+
+// ECSVolumeDriver contains the methods for volume drivers to implement
+type ECSVolumeDriver interface {
```

Comment:
Currently we have a VolumeDriver that calls more VolumeDrivers. IMO, I'd rather call this abstraction layer something else. MountableVolume, MountHelper, Mount, etc

Reply:
we already have a mount helper as part of the struct (`EFSVolumeDriver` ) that implements this interface, hence didn't want to call this MountHelper. To me, it makes sense that the main volume driver is composed of multiple drivers for different volume types. How about calling the main one `AmazonECSVolumePlugin` instead of `AmazonECSVolumeDriver` ?